File: e2e/cypress-javascript/interactions/pages/LanguageSwitcherDropdown.js
class LanguageSwitcherDropdown {
  get dropdown() {
    return cy.get('[data-testid="language-switcher"]');
  }

  get dropdownValue() {
    return this.dropdown.find('span.p-dropdown-label');
  }

  get items() {
    return cy.get('.p-dropdown-item');
  }
}

module.exports = { LanguageSwitcherDropdown };

File: e2e/capybara-ruby/interactions/pages/language_switcher_dropdown.rb
# frozen_string_literal: true

require_relative '../../support/capybara_helpers'

class LanguageSwitcherDropdown
  include CapybaraHelpers
  def dropdown
    find('[data-testid="language-switcher"]')
  end

  def dropdown_value
    dropdown.find('span.p-dropdown-label')
  end

  def items
    find_all('.p-dropdown-item')
  end
end
